### PR TITLE
epic-games fix for Windows 

### PR DESCRIPTION
**Pull request by ccsimplyspolit**
i found fix w capcha by allowing javascript and added new UserAgent

### CODE REVIEW

File: epic-games.js [hunk 1]
```diff
@@ -1,292 +1,294 @@
...
+
+const db = await jsonDb('epic-games.json', {});
+
+if (cfg.time) console.time('startup');
+
+// https://playwright.dev/docs/auth#multi-factor-authentication
+const context = await firefox.launchPersistentContext(cfg.dir.browser, {
+  headless: cfg.headless,
+  viewport: { width: cfg.width, height: cfg.height },
+  userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101 Firefox/127.0', // see replace of Headless in util.newStealthContext. TODO Windows UA enough to avoid 'device not supported'? update if browser is updated?
```
Comment (ccsimplyspolit) [inline epic-games.js:23]: New UserAgent

File: epic-games.js [hunk 2]
```diff
@@ -1,292 +1,294 @@
...
+  // userAgent firefox (docker): Mozilla/5.0 (X11; Linux aarch64; rv:109.0) Gecko/20100101 Firefox/115.0
+  locale: 'ru-RU', // ignore OS locale to be sure to have english text for locators
+  recordVideo: cfg.record ? { dir: 'data/record/', size: { width: cfg.width, height: cfg.height } } : undefined, // will record a .webm video for each page navigated; without size, video would be scaled down to fit 800x800
+  recordHar: cfg.record ? { path: `data/record/eg-${filenamify(datetime())}.har` } : undefined, // will record a HAR file with network requests and responses; can be imported in Chrome devtools
+  handleSIGINT: false, // have to handle ourselves and call context.close(), otherwise recordings from above won't be saved
+  javaScriptEnabled: true,
+  // user settings for firefox have to be put in $BROWSER_DIR/user.js
+  args: [ // https://wiki.mozilla.org/Firefox/CommandLineOptions
+    // '-kiosk',
+    '-purgecaches'
```
Comment (ccsimplyspolit) [inline epic-games.js:34]: purge cache

File: epic-games.js [hunk 3]
```diff
@@ -1,292 +1,294 @@
...
+  viewport: { width: cfg.width, height: cfg.height },
+  userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101 Firefox/127.0', // see replace of Headless in util.newStealthContext. TODO Windows UA enough to avoid 'device not supported'? update if browser is updated?
+  // userAgent firefox (macOS): Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:106.0) Gecko/20100101 Firefox/106.0
+  // userAgent firefox (docker): Mozilla/5.0 (X11; Linux aarch64; rv:109.0) Gecko/20100101 Firefox/115.0
+  locale: 'ru-RU', // ignore OS locale to be sure to have english text for locators
+  recordVideo: cfg.record ? { dir: 'data/record/', size: { width: cfg.width, height: cfg.height } } : undefined, // will record a .webm video for each page navigated; without size, video would be scaled down to fit 800x800
+  recordHar: cfg.record ? { path: `data/record/eg-${filenamify(datetime())}.har` } : undefined, // will record a HAR file with network requests and responses; can be imported in Chrome devtools
+  handleSIGINT: false, // have to handle ourselves and call context.close(), otherwise recordings from above won't be saved
+  javaScriptEnabled: true,
```
Comment (ccsimplyspolit) [inline epic-games.js:30]: allow javascript

**Comment by vogler** [comment]
Looks like you changed all line endings which is why it shows the whole file as changed unless you disable whitespace diffing: https://github.com/vogler/free-games-claimer/pull/337/files?diff=split&w=1

**Comment by ccsimplyspolit** [comment]
looks like all work for me

Unreal engine / Gog / Epic Games /

**Comment by vogler** [comment]
To fix  #338 and the error `Please turn JavaScript on and reload the page`, updating `userAgent` was enough.
Haven't tested the automated login in some time. Closing this since the other changes here are not needed, but others were since they changed the login form again. Thanks!

**Pull request closed without merging**